build/documentation/generate-typedocs.js

```diff
@@ -0,0 +1,198 @@
+/**
+ * Utility for generating Typedoc API documentation and placing it in the docs/en/packages folder so it can be hosted by
+ * Guthub pages and viewed through Docusaurus
+ * 
+ * Usage :run node build/documentation/generate-typedocs.js OR
+ * Usage (dry-run): run node build/documentation/generate-typedocs.js --dry-run
+ */
+
+const path = require("path");
+const fs = require("fs");
+const glob = require("glob");
+const os = require('os');
+const { exec } = require('child_process');
+const { spawn } = require("child_process");
+
+const rootDir = path.resolve(process.cwd());
+const srcDir = "packages/*";
+const destDir = path.join("docs", "en", "packages");
+
+var copyReadmeScriptPath = path.join('build', 'documentation', 'copy-package-readme.js');
+var dryRun = false;
+
+const excludedPackages = [
+    "fast-animation",
+    "fast-browser-extensions",
+    "fast-development-site-react",
+    "fast-permutator",
+    "fast-tslint-rules"
+];
+
+/**
+ * Determine if a dry run will be executed based off --dry-run argument being present
+ * if an invalid third parameter is entered, the application will exit
+ */
+process.argv.forEach(function (val, index) {
+
+    var validArg = true;
+
+    if (index == 2) {
+        val === "--dry-run" ? dryRun = true : validArg = false;
+    }
+
+    if (!validArg) {
+        console.log('\x1b[31m%s\x1b[0m', 'Invalid argument used. To perform a dry-run use --dry-run');
+        process.exit(1);
+    }
+
+});
+
+/**
+ * Run the copy-package-readme.js script, then this one if successful
+ * To prevent new links constantly being appended to the readme every 
+ * time the script is run, copyPackageReadme.js is run to regenerate 
+ * doc/en/packages files to their original form with the needed header
+ */
+
+var proc = dryRun ? exec(`node ${copyReadmeScriptPath} --dry-run`) : exec(`node ${copyReadmeScriptPath}`);
+
+proc.stdout.on('data', function(data) {
+    process.stdout.write(data);
+});
+
+proc.on('error', (err) => {
+    console.log(`\x1b[31m%s\x1b[0m`, err);
+});
+
+proc.on('close', function(code) {
+    console.log(`\x1b[32m%s\x1b[0m`, `copy-package-readme.js completed successfully.`);
+    execute();
+});
+
+/**
+ * Generate TypeDocs for each package
+ */
+function execute() {
+
+    if (dryRun) {
+        console.log(`In docs/en/packages/, this script would...`);
+    } else {
+        console.log(`Generating API documentation using TypeDoc...`);
+    }
+
+    const packages = path.resolve(rootDir, srcDir);
+
+    glob(packages, {realpath:true}, function(error, srcFiles) {
+       
+        srcFiles.forEach((srcPath) => {    
+            
+            var valid = true;
+
+            var packageName = srcPath
+                .split(path.sep)
+                .pop();
+
+            excludedPackages.forEach((excludedPackageName) => {
+                if (packageName == excludedPackageName) {
```

Comment:
Curious - are you intentionally not using strict equality here?

Reply:
No, I'll have to fix it